internal/service/container/kill.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package container
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	cerrdefs "github.com/containerd/errdefs"
+	ncTypes "github.com/containerd/nerdctl/pkg/api/types"
+
+	"github.com/runfinch/finch-daemon/pkg/errdefs"
+)
+
+func (s *service) Kill(ctx context.Context, cid string, options ncTypes.ContainerKillOptions) error {
+	cont, err := s.getContainer(ctx, cid)
+	if err != nil {
+		if cerrdefs.IsNotFound(err) {
+			return errdefs.NewNotFound(err)
+		}
+		return err
+	}
+	status := s.client.GetContainerStatus(ctx, cont)
+	if status != containerd.Running {
+		return errdefs.NewConflict(fmt.Errorf("container is not running id: %s", cid))
```

Comment:
API error response is slightly different in the spec https://docs.docker.com/reference/api/engine/version/v1.43/#tag/Container/operation/ContainerKill
```
Container %s is not running
```